lib/common/entropy_common.c

```diff
@@ -137,14 +138,15 @@ size_t FSE_readNCount_body(short* normalizedCounter, unsigned* maxSVPtr, unsigne
              * because we already memset the whole buffer to 0.
              */
 
-            if ((ip <= iend-7) || (ip + (bitCount>>3) <= iend-4)) {
+            if (LIKELY(ip <= iend-7) || (ip + (bitCount>>3) <= iend-4)) {
                 assert((bitCount >> 3) <= 3); /* For first condition to work */
                 ip += bitCount>>3;
                 bitCount &= 7;
-                bitStream = MEM_readLE32(ip) >> bitCount;
             } else {
-                bitStream >>= 2;
+                bitCount -= (int)(8 * (iend - 4 - ip));
+                ip = iend - 4;
             }
+            bitStream = MEM_readLE32(ip) >> bitCount;
```

Comment:
Some remaining corner case :
`../../lib/common/entropy_common.c:149:42: runtime error: shift exponent 40 is too large for 32-bit type 'U32' (aka 'unsigned int')`

Reply:
Fixed, and also removed the `& 31` from the common case.